src/components/toast/global_toast_list.tsx

```diff
@@ -48,14 +52,28 @@ export interface EuiGlobalToastListProps extends CommonProps {
    * Determines which side of the browser window the toasts should appear
    */
   side?: ToastSide;
+  /**
+   * At this threshold, a "Clear all" button will display at the bottom of the toast list
+   * that allows users to dismiss all toasts in a single click.
+   *
+   * Defaults to `3`. Set to `0` to disable the button entirely.
+   */
+  showClearAllButtonAt?: number;
+  /**
+   * Optional callback that fires when a user clicks the "Clear all" button.
+   * Ignored if `showClearAll` is not `true`.
+   */
+  onClearAllToasts?: () => void;
 }
 
 export const EuiGlobalToastList: FunctionComponent<EuiGlobalToastListProps> = ({
   className,
   toasts = [],
   dismissToast: dismissToastProp,
   toastLifeTimeMs,
+  onClearAllToasts,
   side = 'right',
+  showClearAllButtonAt = CLEAR_ALL_TOASTS_THRESHOLD_DEFAULT,
```

Comment:
TBH I think we can just lose this const entirely
```suggestion
  showClearAllButtonAt = 3,
```

Reply:
I retained the const so a reference can be made to the default threshold in tests for the dismiss button behaviour

Reply:
Gotcha!